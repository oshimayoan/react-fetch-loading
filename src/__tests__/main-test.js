// @flow

import React from 'react';
import loadingInjector from '../main';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

describe('loadingInjector', () => {
  const Component = (props: Object) => {
    return <h1 {...props}/>;
  };
  let WrappedComponent = loadingInjector('isFetching')(Component);

  it('should return wrapped component', () => {
    let isFetching = false;
    let container = shallow(
      <WrappedComponent isFetching={isFetching} />
    );
    expect(toJson(container)).toMatchSnapshot();
  });

  it('should return loading component', () => {
    let isFetching = true;
    let container = shallow(
      <WrappedComponent isFetching={isFetching} />
    );
    expect(toJson(container)).toMatchSnapshot();
  });
});
