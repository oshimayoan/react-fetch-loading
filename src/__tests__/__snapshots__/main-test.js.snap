// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loadingInjector should return loading component 1`] = `
<div
  style={
    Object {
      "display": "block",
      "textAlign": "center",
    }
  }
>
  <LoadingIndicator
    color={
      Object {
        "alpha": 0.3843137254901961,
        "blue": 0,
        "green": 0,
        "red": 0,
      }
    }
    fadeSteps={6}
    fadeTo={0.3163265306122449}
    segmentLength={10}
    segmentWidth={5}
    segments={10}
    spacing={2}
  />
</div>
`;

exports[`loadingInjector should return wrapped component 1`] = `
<Component
  isFetching={false}
/>
`;
